lib/conjur/authenticator.rb

```diff
@@ -0,0 +1,76 @@
+module Conjur
+  # Keeps a fresh Conjur access token in a named file by re-authenticating as needed.
+  class Authenticator
+    TOKEN_LIFESPAN = ( ENV['CONJUR_TOKEN_LIFESPAN'] || 5 * 60 ).to_i.seconds
+    DELAY = ( ENV['CONJUR_TOKEN_REFRESH_DELAY'] || 10 ).to_i.seconds
+
+    attr_reader :authenticate, :filename
+
+    # +authenticate+ should be a proc that authenticates with Conjur and returns an 
+    # access token as a Hash.
+    def initialize authenticate, filename
+      @authenticate = authenticate
+      @filename = filename
+    end
+
+    class << self
+      def default_filename
+        "/etc/conjur-auth/token"
+      end
+
+      # Check the token every +DELAY+ seconds and refresh it if it's out of date. 
+      def run authenticate:, filename: default_filename
```

Comment:
Does the token have an `expires_at` timestamp?  Instead of polling for elapsed time, it could just run a re-authenticate method once the `expires_at` time has passed.